components/gallery/PhotoLibrary.js

```diff
@@ -7,29 +7,46 @@ import {
   Text,
   View,
   ScrollView,
-  CameraRoll
+  CameraRoll,
+  Platform,
+  PermissionsAndroid
 } from "react-native";
 import Photo from "./Photo";
 
 const window = Dimensions.get("window");
 
 class PhotoLibrary extends Component {
   state = {
-    photos: []
+    photos: [],
+    message: "Loading..."
   };
 
   // TODO: maybe all children should just do this by default HOC???
   async componentDidMount() {
     this.props.shouldRenderBackButton(true, this.props.history);
-    this.setState(await this.accessCameraRoll());
+    if (Platform.OS === "android") {
```

Comment:
This is fine for now but both this and the other permission check would ideally be composable (HOC, hook, or just an exported function) for reusability and readability in this component 🤷‍♂️

Reply:
:100: